src/ring_jwt_middleware/core.clj

```diff
@@ -333,9 +333,59 @@
              ['_ `(check-identity-filter! ~authorized (:identity ~'+compojure-api-request+))]))
 
 
+(defn to-scope-repr
+  [txt]
+  (let [[path access] (str/split txt #":")]
+    {:path (str/split path #"/")
+     :access (case access
+               "read"  #{:read}
+               "write" #{:write}
+               "rw"    #{:read :write}
+               nil     #{:read :write}
+               (ring.util.http-response/unauthorized!
+                {:msg "bad access part in the scope, must be read or nothin."}))}))
+(defn sub-list
+  [req-list scope-path-list]
+  (let [n (count scope-path-list)]
+    (= (take n req-list) scope-path-list)))
+
+(defn match-access
+  [required-access access]
+  (clojure.set/subset? required-access access))
+
+(defn match-scope
+  [required-scope scope]
+  (and (match-access (:access required-scope) (:access scope))
+       (sub-list (:path required-scope) (:path scope))))
+
+
+(defn accepted-by-scopes
+  "scopes should be strings.
+  if none of the string contains a `/` nor a `:`.
+  It works as is a subset of.
+
+  :scopes #{\"foo\" \"bar\"}
+  only people with scopes which are super sets of
+  #{\"foo\" \"bar\"}
+  will be allowed to use the route.
+
+  scopes are considered as path with read/write access.
+  so \"foo/bar/baz:read\" is a sub-scope of \"foo\"
+  and of \"foo:read\".
+
+  So the more precise rule of access is.
+  All mandatory scopes must be sub-scopes of at least one user scopes.
+  "
+  [required scopes]
```

Comment:
Why not compare scopes with the same format (after applying `to-scope-repr`)?

Reply:
`required` is pre-formatted at compilation time (while defining the route) to prevent doing the same operation other and other again.

While `scopes` are provided at runtime by the middleware.

So it's just a minimal optimisation.